From b3e8cba4227182fd9e47bbf3b1b0ae7cb769370e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Thu, 26 Sep 2024 05:18:24 +0200
Subject: [PATCH] build: Use appstream for appdata validation test

appstream-glib is under heavy maintenance mode and recommends using
appstream instead.
---
 data/meson.build                       | 8 ++++----
 data/nl.hjdskes.gcolor3.appdata.xml.in | 2 +-
 2 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/data/meson.build b/data/meson.build
index dc78e4a..c33b6cb 100644
--- a/data/meson.build
+++ b/data/meson.build
@@ -32,12 +32,12 @@ appdata_file = i18n.merge_file(
   install: true,
   install_dir: join_paths(datadir, 'metainfo')
 )
-appstream_util = find_program('appstream-util', required: false)
-if appstream_util.found()
+appstreamcli = find_program('appstreamcli', required: false)
+if appstreamcli.found()
   test(
-    'validate-appdata', appstream_util,
+    'validate-appdata', appstreamcli,
     args: [
-      'validate-relax', '--nonet', appdata_file.full_path()
+      'validate', '--no-net', '--explain', appdata_file.full_path()
     ]
   )
 endif
diff --git a/data/nl.hjdskes.gcolor3.appdata.xml.in b/data/nl.hjdskes.gcolor3.appdata.xml.in
index fb82cb9..74c682a 100644
--- a/data/nl.hjdskes.gcolor3.appdata.xml.in
+++ b/data/nl.hjdskes.gcolor3.appdata.xml.in
@@ -43,7 +43,7 @@
   <developer_name>Christopher Davis</developer_name>
   <update_contact>christopherdavis_at_gnome.org</update_contact>
   <translation type="gettext">gcolor3</translation>
-  <!-- Validate with `appstream-util validate nl.hjdskes.gcolor3.appdata.xml.in`. -->
+  <!-- Validate with `appstreamcli validate nl.hjdskes.gcolor3.appdata.xml.in`. -->
   <releases>
     <release version="2.4.0" date="2020-08-24">
       <description>
-- 
GitLab

